### PR TITLE
variable: refactor session/global validation (part 7)

### PR DESCRIPTION
**Pull request by morgo**
### What problem does this PR solve?

This is a followup to https://github.com/pingcap/tidb/pull/23383  https://github.com/pingcap/tidb/pull/23566 https://github.com/pingcap/tidb/pull/23719 https://github.com/pingcap/tidb/pull/23777 https://github.com/pingcap/tidb/pull/23842 https://github.com/pingcap/tidb/pull/23855

### What is changed and how it works?

What's Changed:

Sysvar set is moved to the struct so that the large switch statement can eventually be removed.

### Related changes

- Earlier PRs

### Check List 

Tests 

- Unit test (covered by existing tests)

Side effects

- Harder cherry picks
- Slightly more verbose code in some instances

### Release note 

- No release note

### CODE REVIEW

File: sessionctx/variable/session.go [hunk 1]
```diff
@@ -1372,97 +1372,6 @@ func (s *SessionVars) ClearStmtVars() {
...
-			if !TiDBOptOn(skipIsolationLevelCheck) || err != nil {
-				return returnErr
-			}
-			// SET TRANSACTION ISOLATION LEVEL will affect two internal variables:
-			// 1. tx_isolation
-			// 2. transaction_isolation
-			// The following if condition is used to deduplicate two same warnings.
-			if name == "transaction_isolation" {
-				s.StmtCtx.AppendWarning(returnErr)
-			}
```
Comment (djshow832) [inline sessionctx/variable/session.go:1392]: Why remove this part?

Comment (xhebox) [reply sessionctx/variable/session.go:1392]: I think that is an always-false branch. I've blamed the git history, but found that it is still an always-false condition. It looks like a copy&paste mistake. Check [this](https://github.com/pingcap/tidb/commit/28e62940daf1816a98d9aaca3fb6bf3b40f4759f).

Comment (morgo) [reply sessionctx/variable/session.go:1392]: It's a switch based on `name` and this is under case `TxnIsolationOneShot`. So the branch for `name == "transaction_isolation"` is always false.

**Pull request merged**